src/gl2d/file.ts

```diff
@@ -0,0 +1,151 @@
+import { type Gl2DNode } from './node';
+import { type Gl2DResource } from './resources';
+
+/**
+ * Represents the root file structure for a GL2D scene.
+ *
+ * This is the top-level container that defines a complete GL2D scene file,
+ * including metadata, scene definitions, node hierarchy, and all resources
+ * required for rendering.
+ * @example
+ * ```typescript
+ * const sceneFile: Gl2DFile = {
+ *   asset: {
+ *     version: "1.0",
+ *     generator: "gl2d-example-generator"
+ *   },
+ *   scene: 0, // Active scene index
+ *   scenes: [
+ *     {
+ *       name: "MainScene",
+ *       nodes: [0, 1] // References to nodes array
+ *     }
+ *   ],
+ *   nodes: [
+ *     {
+ *       type: "container",
+ *       name: "Player",
+ *       children: [1],
+ *       transform: { position: [200, 150] },
+ *     }
+ *   ],
+ *   resources: [
+ *     { type: "texture", src: "/textures/hero.png" }
+ *   ]
+ * };
+ * ```
+ * @category gl2d
+ * @standard
+ */
+export interface Gl2DFile extends Gl2DExtension
```

Comment:
Nit about name casing. I think "GL2DFile" might be better. Kinda aligns with HTMLText and less ambiguous lowercase L. But I realize there are a lot of instances of this thing in the PR.